packages/studio-plugin/src/types/messages.ts

```diff
@@ -63,4 +71,19 @@ export type ResponseEventMap = {
   [MessageID.GenerateTestData]: (BaseResponse | ErrorResponse) & {
     mappingJson: Record<string, string[]>;
   };
+  [MessageID.GetEntities]:
+    | ErrorResponse
+    | (BaseResponse & {
```

Comment:
Is there a better way to represent these types?

Reply:
The only way I could think of to simplify the types is to allow any errors that are thrown to propagate up to `studio` via `FatalErrorResponse` instead of catching them in the listener handlers and returning an `ErrorResponse`. But, this would make the invoking code we'll add in `studio` more complicated because we'll have to catch the errors there. If you have a strong preference though, I can switch to that

Reply:
Maybe we can have some kind of generic?

Reply:
I thought about a generic, but I was having a hard time coming up with a name that made sense. We have two types of responses with data right now, one where the `ErrorResponse` also returns the data (e.g. for `GenerateTestData`) and one where it doesn't (e.g. endpoints from this PR). What do you think of `ErrorOrResponseWithData` for the latter?

Reply:
Maybe `ManagementApiResponse`? It's a little specific but I think we don't have to be super generic here (which I guess is a little ironic?)